File: app/models/post.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: posts
#
#  id             :bigint           not null, primary key
#  content        :text             not null
#  team_id        :bigint           not null
#  author_id      :bigint           not null
#  lesson_id      :bigint
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#  discarded_at   :datetime
#  likes_count    :integer          default(0), not null
#  comments_count :integer          default(0), not null
#
class Post < ApplicationRecord
  include Discard::Model

  belongs_to :team
  belongs_to :author, class_name: 'User', inverse_of: :posts
  belongs_to :lesson, optional: true
  has_many :comments, as: :commentable, dependent: :destroy
  has_many :post_likes, dependent: :destroy

  validates :content, presence: true
end

File: frontend/students/models/post.ts
export interface IPostApiResponse {
  post: IPost;
}

export interface IPostsApiResponse {
  posts: Array<IPost>;
}

export interface IPost {
  id: number;
  content: string;
  lessonId: number;
  commentsCount: number;
  likesCount: number;
  createdAt?: string;
  author: IPostAuthor;
  lesson?: IPostLesson;
  comments?: Array<IPostComment>;
}

export interface IPostAuthor {
  avatarUrl: string | null;
  fullName: string;
}

export interface IPostLesson {
  title: string;
  authorName: string;
  avatarUrl: string | null;
}

export interface IPostComment {
  authorName: string;
  avatarUrl: string | null;
  content: string;
  createdAt: string;
}
